File: CmsEngine/Data/ViewModels/DataTableViewModel.cs
using System.Collections.Generic;

namespace CmsEngine.Data.ViewModels
{
    public class DataTableViewModel
    {
        public int RecordsTotal { get; set; }

        public int RecordsFiltered { get; set; }

        public List<string> Columns { get; set; }

        public List<DataItem> Rows { get; set; }
    }

    public class DataItem
    {
        public List<DataProperty> DataProperties { get; set; }
    }

    public class DataProperty
    {
        public string DataType { get; set; }
        public string DataContent { get; set; }
    }
}

File: CmsEngine.Ui.Angular/ClientApp/app/models/datatable-viewmodel.ts
export class DataTableViewModel {
  public recordsTotal: number;
  public recordsFiltered: number;
  public columns: string[];
  public rows: Array<DataItem>;
}

export class DataItem {
  public dataProperties: Array<DataProperty>;
}

export class DataProperty {
  public dataType: string;
  public dataContent: string;
}
